### PR TITLE
[DeepSparse Evaluator API][UI] `EvaluationRegistry.resolve` implementation

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Feature Description

Introduces the `resolve` class method for `EvaluationRegistry`.
Additional changes:
- `evaluate` method is now responsible for creating the `model: Union[Pipeline, Any]` from `target:str`

## Testing
- updated the tests
- ran successfully all the tests that require `lm-eval` as a dependency
- ran example API and CLI commands from internal docs (`[Documentation] Deepsparse Evaluation API`)

### CODE REVIEW

File: src/deepsparse/evaluation/registry.py [hunk 1]
```diff
@@ -14,21 +14,60 @@
...
+        integration: Optional[str] = None,
+    ) -> Callable[..., "Result"]:  # noqa: F821
+        """
+        Chooses an evaluation function from the registry based on the target,
+        datasets and integration.
+
+        If integration is specified, attempts to load the evaluation function
+        from the registry.
+        """
+        from deepsparse.evaluation.utils import resolve_integration
```
Comment (dbogunowicz) [inline src/deepsparse/evaluation/registry.py:52]: To avoid circular import dependency.

File: src/deepsparse/evaluation/utils.py [hunk 2]
```diff
@@ -13,14 +13,51 @@
...
+    "args_to_dict",
+    "resolve_integration",
+]
+
+
+def resolve_integration(
+    model: Union[Pipeline, PreTrainedModel], datasets: Union[str, List[str]]
+) -> Union[str, None]:
+    """
+    Given a model and dataset, inferr the name of the evaluation integration
```
Comment (dbogunowicz) [inline src/deepsparse/evaluation/utils.py:35]:
```suggestion
    Given a model and dataset, infer the name of the evaluation integration
```

File: src/deepsparse/evaluation/utils.py [hunk 3]
```diff
@@ -13,14 +13,51 @@
...
+    if is_model_llm(model):
+        return "llm-evaluation-harness"
+    return None
+
+
+def is_model_llm(model: Any) -> bool:
+    """
+    Checks if the model is a Large Language Model (LLM) type model.
+    """
+    if isinstance(model, Pipeline):
```
Comment (dbogunowicz) [inline src/deepsparse/evaluation/utils.py:55]: Maybe we can make it nicer with instance checking?

**Review by bfineran — APPROVED** [review]

**Comment by bfineran** [comment]
see failures - looks like bad name in tests, potentially bad rebase

**Review by bfineran — APPROVED** [review]

**Pull request merged**